hvac/tests/integration_tests/api/auth/test_mfa.py

```diff
@@ -0,0 +1,240 @@
+                )
+            self.assertIn(
+                member=exception_message,
+                container=str(cm.exception),
+            )
+        else:
+            expected_status_code = 204
+            configure_response = self.client.mfa.configure_duo_behavior(
+                mount_point=mount_point,
+                push_info=push_info,
+                user_agent=user_agent,
+                username_format=username_format,
+            )
+            self.assertEqual(
+                first=expected_status_code,
+                second=configure_response.status_code
+            )
+
+            read_config_response = self.client.mfa.read_duo_behavior_configuration(
+                mount_point=mount_point,
+            )
+            self.assertEqual(
+                first=push_info,
+                second=read_config_response['data']['push_info']
+            )
+
+    @parameterized.expand([
+        ('read configured path', TEST_AUTH_PATH),
+    ])
+    def test_read_duo_behavior_configuration(self, test_label, mount_point, add_configuration=True):
+        if add_configuration:
+            self.client.mfa.configure(
+                mount_point=mount_point,
+            )
+
+        response = self.client.mfa.read_duo_behavior_configuration(
+            mount_point=mount_point,
+        )
+        self.assertIn(
+            member='data',
+            container=response,
+        )
+
+    @parameterized.expand([
+        ('login success',),
+        ('login without duo access configured', False, exceptions.InvalidRequest, "Duo access credentials haven't been configured."),
+    ])
+    def test_login_with_mfa(self, test_label, configure_access=True, raises=None, exception_message=''):
+        username = 'somedude'
+        password = 'myverygoodpassword'
+
+        self.client.mfa.configure(
+            mount_point=TEST_AUTH_PATH,
+        )
+        if configure_access:
+            self.client.mfa.configure_duo_access(
+                mount_point=TEST_AUTH_PATH,
+                host='localhost:{port}'.format(port=self.mock_server_port),
+                integration_key='an-integration-key',
+                secret_key='valid-secret-key'
+            )
+
+        self.client.create_userpass(
+            username=username,
+            password=password,
+            policies=['defaut'],
+            mount_point=TEST_AUTH_PATH,
+        )
+        if raises:
+            with self.assertRaises(raises) as cm:
+                self.client.auth_userpass(
+                    username=username,
+                    password=password,
+                    mount_point=TEST_AUTH_PATH,
+                )
+            self.assertIn(
+                member=exception_message,
+                container=str(cm.exception),
+            )
+        else:
+            raise SkipTest('Unable to fully test login flow (easily) without working TLS for mock Duo server')
+            login_response = self.client.auth_userpass(
```

Comment:
having code under an unconditional `raise` smells funny.  not hurting anything, but maybe comment out the code code under raise? meh.